li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/producer/LiKafkaProducerImpl.java

```diff
@@ -160,6 +167,15 @@ private LiKafkaProducerImpl(LiKafkaProducerConfig configs,
                              Auditor<K, V> auditor) {
     // Instantiate the open source producer, which always sents raw bytes.
     _producer = new KafkaProducer<>(configs.originals(), new ByteArraySerializer(), new ByteArraySerializer());
+    // TODO: This hack remains until bounded flush is added to apache/kafka
+    Method producerSupportsBoundedFlush;
+    try {
+      producerSupportsBoundedFlush = _producer.getClass().getMethod("flush", long.class, TimeUnit.class);
+    } catch (NoSuchMethodException e) {
+      LOG.warn("LiKafkaProducerImpl does not support bounded flush.");
```

Comment:
Maybe the message should say "Wrapped Kafka producer does not have a time bounded flush method."

Reply:
i'd actually print the expected method signature not found

Reply:
Ok I will add the method signature.